examples/transfer-collection.rs

```diff
@@ -0,0 +1,126 @@
+//! Example that shows how to create a collection, and transfer it to another
+//! node. It also shows patterns for defining a "Node" struct in higher-level
+//! code that abstracts over these operations with an API that feels closer to
+//! what an application would use.
+//!
+//! Run the entire example in one command:
+//!  $ cargo run --example transfer-collection
+use std::collections::HashMap;
+
+use anyhow::{Context, Result};
+use iroh::{protocol::Router, Endpoint, NodeAddr, Watcher};
+use iroh_blobs::{
+    api::{downloader::Shuffled, Store, TempTag},
+    format::collection::Collection,
+    store::mem::MemStore,
+    BlobsProtocol, Hash, HashAndFormat,
+};
+
+/// Node is something you'd define in your application. It can contain whatever
+/// shared state you'd want to couple with network operations.
+struct Node {
+    store: Store,
+    /// Router with the blobs protocol registered, to accept blobs requests.
+    /// We can always get the endpoint with router.endpoint()
+    router: Router,
+}
+
+impl Node {
+    async fn new() -> Result<Self> {
+        let endpoint = Endpoint::builder().bind().await?;
```

Comment:
I'm skipping `.discovery_n0()` here and using `add_node_addr` directly on the endpoint in the `Node::get_collection` method below. Largely to take a highligher to the `add_node_addr` discussion we've been having.

@rklaehn: the store has a connection pool, but I can't seem to get at it? Seems the connection pool _relies_ on me being able to plumb addresses into the endpoint, is that right?